ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/map.view.js

```diff
@@ -263,10 +264,16 @@ module.exports = Marionette.LayoutView.extend({
   },
   saveAsHome: function() {
     const boundingBox = this.map.getBoundingBox()
-    user
-      .get('user')
-      .get('preferences')
-      .set('mapHome', boundingBox)
+    const userPreferences = user.get('user').get('preferences')
+    userPreferences.once('sync', () =>
+      announcement.announce({
+        title: 'New Home Set',
+        message: 'Home location has been set!',
+        type: 'success',
+      })
+    )
+
+    userPreferences.set('mapHome', boundingBox)
```

Comment:
Is there a way to check that this function didn't result in an error before showing the above message? https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js#L214

Reply:
I believe, based on my limited knowledge of Backbone, that the `sync` event will only fire if successful so the message should only be displayed only in that case as well. I'll try to double check though.

Reply:
I looked into this a little and basically there's such a small chance that saving a preference will result in an invalid sync response as to be next to impossible.